napari/components/layerlist.py

```diff
@@ -327,3 +328,36 @@ def save(
             return []
 
         return save_layers(path, layers, plugin=plugin)
+
+    def selection_context(self):
+        return {k: v(self.selection) for k, v in CONTEXT_KEYS.items()}
+
+
+# Each key in this list is "usable" as a variable name in the the "enable_when"
+# and "show_when" expressions of the napari.layers._layer_actions.LAYER_ACTIONS
+#
+# each value is a function that takes a LayerList.selection, and returns
+# a value. LayerList.selection_context uses this dict to generate a concrete
+# context object that can be passed to the
+# `qt_action_context_menu.QtActionContextMenu` method to update the enabled
+# and/or visible items based on the state of the layerlist.
+
+CONTEXT_KEYS = {
```

Comment:
Love this, but can we make it private, `_CONTEXT_KEYS`. I suspect we might refine some of these names, but don't want to get hung up on them now. If you think important to be public we can talk through more of them. The most confusing one for me is `active_shape`, not quite clear what that one is just by reading it

Reply:
Sure, private is fine.

Will change to `active_layer_shape`